enhancements/sdk-generate-commands.md

```diff
@@ -0,0 +1,411 @@
+[bundle][bundle-format] and [package manifests][package-manifests-format] format
+generation scenarios with a plethora of flags. The number of flags and confusing
+command name (`generate csv` manages higher-level formats) creates non-ideal UX.
+This proposal details new workflows for bundle and package manifests formats
+for old and [new][ep-kb-integration] `operator-sdk`-build Operator projects.
+
+To clearly define what old and new layouts mean:
+* Old project layouts: scaffolded with `operator-sdk new`
+* New project layouts: scaffolded with `operator-sdk init`
+
+We plan to improve the developer experience for operator authors using Operator SDK by providing:
+1. One `operator-sdk` subcommand _per format_ for generating manifests and metadata.
+1. Interactive-by-default commands to get UI information for packaged manifests and metadata.
+
+[bundle-format]:https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md#bundle-manifest-format
+[package-manifests-format]:https://github.com/operator-framework/operator-registry/tree/v1.5.3/#manifest-format
+[ep-kb-integration]:https://github.com/operator-framework/operator-sdk/blob/master/proposals/kubebuilder-integration.md
+
+## Motivation
+
+Currently, the `operator-sdk generate csv` command is used for generating CSVs,
+which scaffolds either a bundle or package manifests format and writes a
+[`ClusterServiceVersion` (CSV)][doc-csv] manifest to disk depending on the format selected.
+This process requires a developer to manually edit the CSV when any required
+field is missing, or if they want to add certain pieces of [UI metadata][doc-csv-metadata].
+Furthermore selecting the format is non-obvious; adding to the confusion is
+the fact that the operator-framework has no plans to deprecate the package manifests format
+in the near future, so having documentation for both formats in one command is confusing.
+
+[doc-csv]:https://github.com/operator-framework/operator-lifecycle-manager/blob/0.14.2/doc/design/building-your-csv.md
+[doc-csv-metadata]:https://github.com/operator-framework/operator-lifecycle-manager/blob/0.14.2/doc/design/building-your-csv.md#your-custom-resource-definitions
+
+## Goals
+
+* Simplify UX for generating bundle and package manifests formats.
+* Delineate bundle and package manifests format generation and usage.
+* _(New project layouts)_ Use a Makefile wherever possible.
+
+## Non-Goals
+
+* Deprecate or remove package manifests support from the Operator SDK.
+* Discuss the `run` subcommand in relation to bundles or package manifests.
+
+## Proposal
+
+There are two parts to this proposal:
+
+1. Add `generate bundle` and `generate packagemanifests` subcommands for both
+old and new project layouts.
+1. _(New project layouts)_ Add Makefile recipes `make bundle` and `make bundle-build`
+that invoke `generate bundle` and `docker build -f bundle.Dockerfile .`, respectively.
+
```

Comment:
IMO `make bundle-build`  does `make bundle` as well. so, why keep both? 

As described in: https://github.com/operator-framework/operator-sdk/pull/2860/files#r431074177

We have now 2 makefile targets which make it a litlle confuse. See:

```
# Build the bundle OCI image
bundle-build: manifests
	$(KUSTOMIZE) build config/bundle | operator-sdk generate bundle -q --manifests --metadata --overwrite --version $(VERSION)
	operator-sdk bundle validate config/bundle
	docker build -f bundle.Dockerfile -t $(BUNDLE_IMG) .

# Generate a bundle directory
bundle: manifests
	$(KUSTOMIZE) build config/bundle | operator-sdk generate bundle -q --manifests --version $(VERSION)
```

Could we not remove the `make bundle` and just keep the `make build-bundle` instead of it which has been used in the example? Would not it make things simpler?

Reply:
Clarification: `make bundle-build` is like running:
```sh
$ make bundle
$ operator-sdk bundle validate config/bundle
$ docker build -t $BUNDLE_IMG -f bundle.Dockerfile .
```

If a user just wants to generate a bundle, they'd run `make bundle`.

Reply:
Why would we force a new generation of the bundle during bundle build doesn't seem necessary.

Also, we must change the wording then above to make clear that the bundle-build is also calling generate bundle via make bundle.

Reply:
It isn't strictly necessary but if you only ever wanted to build bundle images you would use `make bundle-build`. I'm fine with:
```make
bundle: manifests
   # generate bundle
   # validate bundle

bundle-build:
   docker build -f bundle.Dockerfile -t $BUNDLE_IMG .
```

Reply:
@shawn-hurley @camilamacedo86 done